tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,225 @@
+# `.Axes.annotate`   `.text.Annotation`             `.text.Annotation.update_p\
+#                                                   ositions`
+# `.Axes.barh`       `.patches.Rectangle`           `.Rectangle.set_angle`,
+#                                                   `.Rectangle.set_bounds`,
+#                                                   `.Rectangle.set_height`,
+#                                                   `.Rectangle.set_width`,
+#                                                   `.Rectangle.set_x`,
+#                                                   `.Rectangle.set_y`
+#                                                   `.Rectangle.set_xy`
+# `.Axes.fill`       `.patches.Polygon`             `.Polygon.set_xy`
+# `.patches.Circle`  `.patches.Ellipse`             `.Ellipse.set_angle`,
+#                                                   `.Ellipse.set_center`,
+#                                                   `.Ellipse.set_height`,
+#                                                   `.Ellipse.set_width`
+# =================  =============================  ===========================
+#
+# Covering the set methods for all types of artists is beyond the scope of this
+# tutorial but can be found in their respective documentations. An example of
+# such update methods in use for `.Axes.scatter` is as follows.
+
+
+fig, ax = plt.subplots()
+t = np.linspace(-4, 4, 400)
+a, b = 3, 2
+delta = np.pi / 2
+
+scat = ax.scatter(np.sin(a * t[0] + delta), np.sin(b * t[0]), c="b", s=2)
+ax.set_xlim(-1.5, 1.5)
+ax.set_ylim(-1.5, 1.5)
+
+
+def update(frame):
+    # .set_offsets also resets the offset data for the entire collection with
+    # the new values. Therefore, to also carry forward the previously
+    # calculated information, we use the data from the first to the current
+    # frame to set the new offsets.
+    x = np.sin(a * t[:frame] + delta)
+    y = np.sin(b * t[:frame])
+    data = np.stack([x, y]).T
+    scat.set_offsets(data)
+    return (scat,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=400, interval=30)
+plt.show()
+
+
+###############################################################################
+# :class:`~matplotlib.animation.ArtistAnimation`
+# ----------------------------------------------
+#
+# :class:`~matplotlib.animation.ArtistAnimation` can be used
+# to generate animations if there is data stored on various different artists.
+# This list of artists is then converted frame by frame into an animation. For
+# example, when we use `.Axes.barh` to plot a bar-chart, it creates a number of
+# artists for each of the bar and error bars. To update the plot, one would
+# need to update each of the bars from the container individually and redraw
+# them. Instead, `.animation.ArtistAnimation` can be used to plot each frame
+# individually and then stitched together to form an animation. A barchart race
+# is a simple example for this.
+
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
```

Comment:
Needs a seed or else the docs will not be reproducible. We usually use 19680801 or something based on it.

Reply:
Got it. Just saw the docs too. Thanks.